service-commands/scripts/seed.js

```diff
@@ -0,0 +1,194 @@
+    if (!alias && !userId) {
+      throw new Error(
+        'Seed CLI: to set-user you must provide either a user ID or user alias of user to set as active.'
+      )
+    } else {
+      console.log(
+        `Setting active user for seed session to userId ${userId} / alias ${alias}.`.info
+      )
+      const seed = new SeedSession()
+      await seed.setUser({ alias, userId })
+      console.log(
+        `Active user for seed session set to userId ${userId} / alias ${alias}.`.success
+      )
+      process.exit(0)
+    }
+  })
+
+Object.entries(CLI_TO_COMMAND_MAP).forEach(
+  ([cliCommand, { api, description, method, params, onSuccess }]) => {
+    let configuredProgram = program
+      .command(cliCommand)
+      .description(description)
+    const userIdInLibsApiMethodParams = params.some(p => p.name === 'userId')
+    if (!userIdInLibsApiMethodParams) {
+      configuredProgram = configuredProgram
+      .option(
+        '-u, --user-id <number>',
+        'ID of user to set as active when performing action',
+        null
+      )
+    }
+
+    params.forEach(param => {
+      let {
+        name,
+        description,
+        userInputHandler
+      } = param
+      if (!userInputHandler) {
+        userInputHandler = passThroughUserInput
+      }
+      const cliOption = camelToKebabCase(name)
+      const shortOption = cliOption.charAt(0)
+      configuredProgram = configuredProgram.option(
+        `-${shortOption} --${cliOption} <value>`,
+        description,
+        userInputHandler,
+        ''
+      )
+    })
+
+    configuredProgram.action(async opts => {
+      const { userId, ...options } = opts.opts()
+      const seed = new SeedSession()
+      console.log(`Running seed ${cliCommand} with options: ${JSON.stringify(options)}`)
+      let userIdToSet = userId
+      if (!userIdToSet) {
+        // try to set from cache
+        console.log(
+          'No user ID for action specified; trying to get active user from seed local cache...'.info
+        )
+        const activeUser = seed.cache.getActiveUser()
+        if (!activeUser.hedgehogEntropyKey) {
+          throw new Error(
+            `There is no active user in seed local cache. Please rerun the 'seed ${cliCommand}' command with --user-id or -id flag or run 'seed set-user' with alias or id flag first to set active user.`.error
+          )
+        }
+        userIdToSet = activeUser.userId
+      }
+      await seed.setUser({ userId: userIdToSet })
+      console.log(`Calling libs.${api}.${method} to seed...`.info)
+
+      for (const option of Object.entries(options)) {
+        let [optionName, userProvidedValue] = option
+        const match = param => param.name === optionName
+        const defaultHandler = params.find(match).defaultHandler || passThroughUserInput
+        options[optionName] = await defaultHandler(userProvidedValue, seed)
+      }
+      const args = userIdInLibsApiMethodParams
+        ? [userId, ...Object.values(options)]
```

Comment:
this line right now is pretty much just to account for `libs.Playlist.createPlaylist` needing `userId` passed in explicitly... I don't love it, but it is what it is right now lol